sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -1070,22 +1176,25 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemBufferMap(
     ur_map_flags_t mapFlags, size_t offset, size_t size,
     uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
     ur_event_handle_t *phEvent, void **ppRetMap) {
-  UR_ASSERT(hBuffer->MemType == ur_mem_handle_t_::Type::Buffer,
-            UR_RESULT_ERROR_INVALID_MEM_OBJECT);
+  UR_ASSERT(ppRetMap != nullptr, UR_RESULT_ERROR_INVALID_NULL_POINTER);
```

Comment:
I find `UR_ASSER(object, ...)` cleaner than testing against `object != nullptr`.

Reply:
I have always been told that checking  `if (ptr != nullptr)` is easier to read than `if (ptr)` but I guess that it's a matter of opinion I guess. LLVM style guide doesn't seem to say anything about this

Reply:
From a quick glance at the source `assert(pointer)` is more common, not a deal breaker obviously, so feel free to ignore.